### PR TITLE
gRPC: add the ability to use a custom SSL certificate in integration tests

### PR DESCRIPTION
**Pull request by var-const**
This closely mimics the way it is currently done in Objective-C gRPC client by simply using a static string that contains a file path.

### CODE REVIEW

File: Firestore/core/src/firebase/firestore/remote/grpc_connection.mm [hunk 1]
```diff
@@ -52,6 +54,8 @@
 
 }  // namespace
 
+std::string GrpcConnection::test_certificate_path_;
```
Comment (var-const) [inline Firestore/core/src/firebase/firestore/remote/grpc_connection.mm]: I'm not particularly concerned about order of destruction here, especially considering that this is test-only, but feel free to push back. Also, will be happy to discuss alternatives.

File: Firestore/core/src/firebase/firestore/remote/grpc_connection.mm [hunk 2]
```diff
@@ -114,9 +118,31 @@
...
+        database_info_->host(),
+        grpc::SslCredentials(grpc::SslCredentialsOptions()));
+  }
+
+  if (test_credentials_->use_insecure_channel) {
+    return grpc::CreateChannel(database_info_->host(),
+                               grpc::InsecureChannelCredentials());
+  }
+
+  std::fstream cert_file{test_credentials_->certificate_path};
```
Comment (rsgowman) [inline Firestore/core/src/firebase/firestore/remote/grpc_connection.mm]: fstream -> ifstream?

File: Firestore/core/src/firebase/firestore/remote/grpc_connection.mm [hunk 3]
```diff
@@ -181,6 +207,31 @@
...
+    absl::string_view certificate_path, absl::string_view target_name) {
+  HARD_ASSERT(!certificate_path.empty(), "Empty path to test certificate");
+  HARD_ASSERT(!target_name.empty(), "Empty SSL target name");
+
+  if (!test_credentials_) {
+    // Deliberately never deleted.
+    test_credentials_ = new TestCredentials{};
+  }
+
+  test_credentials_->certificate_path = certificate_path.data();
```
Comment (rsgowman) [inline Firestore/core/src/firebase/firestore/remote/grpc_connection.mm]: I don't think this is safe? string_view::data() isn't guaranteed to be null terminated, and I think std::string's operator= assumes the pointer is.

Maybe:
test_credentials_->certificate_path = std::string(cert_path.data(), cert_path.size());

Also immediately below.

File: Firestore/core/src/firebase/firestore/remote/grpc_connection.mm (hunk 2 above)
Comment (var-const) [reply Firestore/core/src/firebase/firestore/remote/grpc_connection.mm]: Done.

File: Firestore/core/src/firebase/firestore/remote/grpc_connection.mm (hunk 3 above)
Comment (var-const) [reply Firestore/core/src/firebase/firestore/remote/grpc_connection.mm]: Done.

**Pull request merged**